.github/workflows/bump-tag-publish.yaml

```diff
@@ -44,91 +44,130 @@ on:
-      - name: Skip version bump merges
-        id: skiptest
-        uses: ./.github/actions/bump-skip
-        with:
-          event-name: ${{ github.event_name }}
-      - name: Bump the tag to a new version
-        if: steps.skiptest.outputs.is-bump == 'no'
-        uses: databiosphere/github-actions/actions/bumper@bumper-0.1.0
-        id: tag
-        env:
-          DEFAULT_BUMP: patch
-          GITHUB_TOKEN: ${{ secrets.BROADBOT_TOKEN }}
-          HOTFIX_BRANCHES: hotfix.*
-          OVERRIDE_BUMP: ${{ steps.controls.outputs.semver-part }}
-          RELEASE_BRANCHES: main
-          VERSION_FILE_PATH: settings.gradle
-          VERSION_LINE_MATCH: "^gradle.ext.tanagraVersion\\s*=\\s*\".*\""
-          VERSION_SUFFIX: SNAPSHOT
+    - name: Set part of semantic version to bump
+      id: controls
+      run: |
+        SEMVER_PART=""
+        CHECKOUT_BRANCH="$GITHUB_REF"
+        if ${{github.event_name == 'push' }}; then
+          SEMVER_PART="patch"
+        elif ${{github.event_name == 'workflow_dispatch' }}; then
+          SEMVER_PART=${{ github.event.inputs.bump }}
+          CHECKOUT_BRANCH=${{ github.event.inputs.branch }}
+        fi
+        echo semver-part=$SEMVER_PART >> $GITHUB_OUTPUT
+        echo checkout-branch=$CHECKOUT_BRANCH >> $GITHUB_OUTPUT
+    - name: Checkout current code
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ steps.controls.outputs.checkout-branch }}
+        token: ${{ secrets.BROADBOT_TOKEN }}
+    - name: Skip version bump merges
+      id: skiptest
+      uses: ./.github/actions/bump-skip
+      with:
+        event-name: ${{ github.event_name }}
+    - name: Bump the tag to a new version
+      if: steps.skiptest.outputs.is-bump == 'no'
+      uses: databiosphere/github-actions/actions/bumper@bumper-0.1.0
+      id: tag
+      env:
+        DEFAULT_BUMP: patch
+        GITHUB_TOKEN: ${{ secrets.BROADBOT_TOKEN }}
+        HOTFIX_BRANCHES: hotfix.*
+        OVERRIDE_BUMP: ${{ steps.controls.outputs.semver-part }}
+        RELEASE_BRANCHES: main
+        VERSION_FILE_PATH: settings.gradle
+        VERSION_LINE_MATCH: "^gradle.ext.tanagraVersion\\s*=\\s*\".*\""
+        VERSION_SUFFIX: SNAPSHOT
 
-      - name: Set up AdoptOpenJDK
-        if: steps.skiptest.outputs.is-bump == 'no'
-        uses: actions/setup-java@v4
-        with:
-          distribution: 'temurin'
-          java-version: 11
-      - name: Cache Gradle packages
-        if: steps.skiptest.outputs.is-bump == 'no'
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~/.gradle/caches
-            ~/.gradle/wrapper
-          key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
-          restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
-      - name: Grant execute permission for gradlew
-        if: steps.skiptest.outputs.is-bump == 'no'
-        run: chmod +x gradlew
+  client-publish:
```

Comment:
Have you been able to run this updated GHA from your branch? Then we can point the SD/AoU teams at a client library that they can test out pre or post merge against our `test` env (running Java 17) if they'd like.

In the past, when working on this action, I think I updated the main branch name hardcoded here from main -> [my branch e.g. java-17] and then opened a PR against my branch and merged it, to check that everything got published/bumped/build/etc correctly.

Reply:
@marikomedlock Tim checked with VUMC - they do not have concerns upgrading the client. Targeting to merge this when you are back from vacation

Reply:
also can confirm that client compile succeeds but I cannot test create via the GH workflow. 
local client published and attached here for vumc testing if needed. 
[tanagra-client.zip](https://github.com/user-attachments/files/16022385/tanagra-client.zip)

cc: @tjennison-google

Reply:
Client compile succeeding is great, but we should test that the publish to Artifactory works.

Can you publish to Artifactory locally? What problem did you hit testing the GHA?